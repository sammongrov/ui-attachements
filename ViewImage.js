import React, { Component } from 'react';
import { View, TouchableOpacity, Dimensions, Alert } from 'react-native';
import PropTypes from 'prop-types';
import { styles } from 'react-native-theme';
import ImageViewer from 'react-native-image-zoom-viewer';
import FastImage from 'react-native-fast-image';
import { Colors } from '@ui/theme_default';
import { Icon, Loading, NavBar, Screen, FontAwesomeIcon } from '@ui/components';
import { Actions } from 'react-native-router-flux';

export default class ViewImage extends Component {
  state = {
    layout: {
      height: Dimensions.get('window').height,
      width: Dimensions.get('window').width,
    },
  };

  _onLayout = (event) => {
    this.setState({
      layout: {
        height: event.nativeEvent.layout.height,
        width: event.nativeEvent.layout.width,
      },
    });
  };

  renderBackButton() {
    const { goBack } = this.props;
    return (
      <TouchableOpacity style={styles.cameraBackButton} onPress={goBack}>
        <Icon name="chevron-left" size={30} color={Colors.ICON_WHITE} width={30} />
      </TouchableOpacity>
    );
  }

  render() {
    const { imageUrl, deleteMessage, showDelete } = this.props;
    const { layout } = this.state;
    return (
      <Screen safeBgColors={['#000', '#000']} onLayout={this._onLayout}>
        <View style={[styles.flex1, { width: layout.width, backgroundColor: '#000' }]}>
          <View
            style={{
              position: 'absolute',
              top: 0,
              left: 0,
              right: 0,
              zIndex: 100,
            }}
          >
            <NavBar
              leftComponent={
                <TouchableOpacity onPress={Actions.pop}>
                  <Icon
                    name="chevron-left"
                    type="material-community"
                    color={Colors.TEXT_WHITE}
                    size={36}
                  />
                </TouchableOpacity>
              }
              rightComponent={
                showDelete && (
                  <TouchableOpacity
                    onPress={() => {
                      Alert.alert(
                        'Delete',
                        'Do you want to delete message?',
                        [
                          { text: 'No', onPress: () => {}, style: 'cancel' },
                          {
                            text: 'Yes',
                            onPress: () => {
                              deleteMessage();
                              Actions.pop();
                            },
                          },
                        ],
                        { cancelable: false },
                      );
                    }}
                  >
                    <FontAwesomeIcon
                      name="trash-o"
                      type="material-community"
                      color={Colors.TEXT_WHITE}
                      size={22}
                    />
                  </TouchableOpacity>
                )
              }
              titleText="Image Preview"
              textStyle={{ color: Colors.TEXT_WHITE }}
              navContainerStyle={{
                backgroundColor: 'rgba(0,0,0,0.7)',
                borderBottomWidth: 0,
              }}
            />
          </View>
          <ImageViewer
            imageUrls={[{ url: imageUrl }]}
            backgroundColor={Colors.BG_BLACK}
            enableImageZoom
            loadingRender={() => (
              <View style={styles.alignJustifyCenter}>
                <Loading />
              </View>
            )}
            renderImage={(props) => <FastImage {...props} />}
            renderIndicator={() => {}}
          />
        </View>
      </Screen>
    );
  }
}

ViewImage.propTypes = {
  imageUrl: PropTypes.string,
  goBack: PropTypes.func.isRequired,
};

ViewImage.defaultProps = {
  imageUrl: '',
};
