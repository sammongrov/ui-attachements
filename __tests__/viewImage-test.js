import React from 'react';
import { View, TouchableOpacity, Dimensions, Alert } from 'react-native';
import ImageViewer from 'react-native-image-zoom-viewer';
import { shallow, configure } from 'enzyme';
import renderer from 'react-test-renderer';
import { Actions } from 'react-native-router-flux';
import Adapter from 'enzyme-adapter-react-16';

import ViewImage from '../ViewImage';

configure({ adapter: new Adapter() });

/* eslint-disable */
jest.mock('react-native-image-zoom-viewer', () => {
  const React = require('React');
  const PropTypes = require('prop-types');
  return class MockImageViewer extends React.Component {
    static propTypes = { children: PropTypes.any };
    static defaultProps = { children: '' };

    render() {
      return React.createElement('ImageViewer', this.props, this.props.children);
    }
    /* eslint-enable */
  };
});

jest.mock('react-native-router-flux', () => ({
  Actions: {
    pop: jest.fn(),
  },
}));

jest.mock('Alert', () => ({
  alert: jest.fn((str1, str2, arr) => {
    arr[0].onPress();
  }),
}));

jest.mock('Dimensions', () => ({
  get: () => ({ width: 720, height: 720 }),
}));

beforeEach(() => {
  jest.resetModules();
});

const imageUrl = 'test_imageUrl';
const deleteMessage = jest.fn();
const showDelete = 'test_showDelete';
const goBack = jest.fn();

it('leftComponent TouchableOpacity is called', () => {
  const rootComp = shallow(
    <ViewImage
      imageUrl={imageUrl}
      deleteMessage={deleteMessage}
      goBack={goBack}
      showDelete={showDelete}
    />,
  );
  const navbar = rootComp.find({ titleText: 'Image Preview' });
  const TO = navbar
    .shallow()
    .find('TouchableOpacity')
    .first();
  TO.props().onPress();

  expect(Actions.pop).toBeCalled();
});

it('ViewImage render correctly', () => {
  const ViewImageTree = renderer
    .create(
      <ViewImage
        imageUrl={imageUrl}
        deleteMessage={deleteMessage}
        goBack={goBack}
        showDelete={showDelete}
      />,
    )
    .toJSON();

  expect(ViewImageTree).toMatchSnapshot();
});

it('rightComponent TouchableOpacity is called', () => {
  const rootComp = shallow(
    <ViewImage
      imageUrl={imageUrl}
      deleteMessage={deleteMessage}
      goBack={goBack}
      showDelete={showDelete}
    />,
  );
  const navbar = rootComp.find({ titleText: 'Image Preview' });
  const TO = navbar
    .shallow()
    .find('TouchableOpacity')
    .last();
  TO.props().onPress();

  expect(showDelete).toBeTruthy();
  expect(Alert.alert).toBeCalled();
});

it('initial state is set', () => {
  const root = shallow(<ViewImage />);
  expect(root.state().layout.width).toBe(720);
  expect(root.state().layout.height).toBe(720);
});

it('_onLayout method is called', () => {
  const event = { nativeEvent: { layout: { height: 360, width: 420 } } };
  const rootComp = shallow(
    <ViewImage
      imageUrl={imageUrl}
      deleteMessage={deleteMessage}
      goBack={goBack}
      showDelete={showDelete}
    />,
  );
  rootComp
    .find('Screen')
    .props()
    .onLayout(event);

  expect(rootComp.state().layout.height).toBe(360);
  expect(rootComp.state().layout.width).toBe(420);
});

it('renderBackButton is called', () => {
  const rootComp = shallow(<ViewImage goBack={goBack} />);
  const instance = rootComp.instance();
  const component = shallow(instance.renderBackButton());

  expect(component).toBeTruthy();
});

it('ImageViewer render correctly', () => {
  const rootComp = shallow(
    <ViewImage
      imageUrl={imageUrl}
      deleteMessage={deleteMessage}
      goBack={goBack}
      showDelete={showDelete}
    />,
  );

  const imageViewer = rootComp.find({ enableImageZoom: true });

  expect(imageViewer).toBeTruthy();

  const viewComponent = shallow(imageViewer.props().loadingRender());
  expect(viewComponent).toBeTruthy();

  const renderImageComp = shallow(imageViewer.props().renderImage());
  expect(renderImageComp).toBeTruthy();

  imageViewer.props().renderIndicator();
  expect(imageViewer.props().renderIndicator).toBeInstanceOf(Function);
});

it('navBar-rightComponent-Alert - Yes', () => {
  jest.mock('Alert', () => ({
    alert: jest.fn((str1, str2, arr) => {
      arr[1].onPress();
    }),
  }));

  const rootComp = shallow(
    <ViewImage
      imageUrl={imageUrl}
      deleteMessage={deleteMessage}
      goBack={goBack}
      showDelete={showDelete}
    />,
  );
  const navbar = rootComp.find({ titleText: 'Image Preview' });
  const TO = navbar
    .shallow()
    .find('TouchableOpacity')
    .last();
  TO.props().onPress();

  expect(deleteMessage).toBeCalled();
  expect(Actions.pop).toBeCalled();
  expect(showDelete).toBeTruthy();
});
