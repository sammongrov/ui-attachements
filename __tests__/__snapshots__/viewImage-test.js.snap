// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewImage render correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4278190080,
      4278190080,
    ]
  }
  locations={
    Array [
      0,
      1,
    ]
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Array [
            undefined,
            Object {
              "backgroundColor": "#000",
              "width": 720,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 100,
            }
          }
        >
          <View
            style={
              Array [
                undefined,
                Object {
                  "backgroundColor": "rgba(0,0,0,0.7)",
                  "borderBottomWidth": 0,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#fff",
                        "fontSize": 36,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  type="material-community"
                >
                  
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  undefined,
                  undefined,
                  null,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "fontFamily": "OpenSans-Regular",
                    },
                    Array [
                      undefined,
                      Object {
                        "color": "#fff",
                      },
                    ],
                  ]
                }
              >
                Image Preview
              </Text>
            </View>
            <View
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#fff",
                        "fontSize": 22,
                      },
                      undefined,
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  type="material-community"
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
        <ImageViewer
          backgroundColor="#000"
          enableImageZoom={true}
          imageUrls={
            Array [
              Object {
                "url": "test_imageUrl",
              },
            ]
          }
          loadingRender={[Function]}
          renderImage={[Function]}
          renderIndicator={[Function]}
        >
          
        </ImageViewer>
      </View>
    </View>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
